controller/engine_controller.go

```diff
@@ -1392,22 +1447,18 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 		ec.eventRecorder.Eventf(e, v1.EventTypeNormal, EventReasonRebuilt,
 			"Replica %v with Address %v has been rebuilt for volume %v", replica, addr, e.Spec.VolumeName)
 
-		autoCleanupSystemGeneratedSnapshot, err := ec.ds.GetSettingAsBool(types.SettingNameAutoCleanupSystemGeneratedSnapshot)
-		if err != nil {
-			log.WithError(err).Errorf("Failed to get %v setting", types.SettingDefinitionAutoCleanupSystemGeneratedSnapshot)
-			return
-		}
-		// If enabled, call SnapshotPurge to clean up system generated snapshot.
+		// If enabled, call SnapshotPurge to clean up system generated snapshot after rebuilding.
 		if autoCleanupSystemGeneratedSnapshot {
 			if err := client.SnapshotPurge(); err != nil {
-				log.WithError(err).Errorf("Failed to start snapshot purge for volume %v", e.Spec.VolumeName)
-				ec.eventRecorder.Eventf(e, v1.EventTypeWarning, EventReasonFailedStartingSnapshotPurge, "Failed to start snapshot purge for volume %v: %v", e.Spec.VolumeName, err)
+				log.WithError(err).Error("Failed to start snapshot purge after rebuilding")
+				ec.eventRecorder.Eventf(e, v1.EventTypeWarning, EventReasonFailedStartingSnapshotPurge, "Failed to start snapshot purge for engine %v and volume %v after rebuilding: %v", e.Name, e.Spec.VolumeName, err)
 				return
 			}
-			logrus.Debugf("Started snapshot purge for volume %v after rebuild is done", e.Spec.VolumeName)
+			logrus.Debug("Started snapshot purge after rebuilding")
```

Comment:
nit: use `log.Debug` to keep the context info around instead of using the general logrus.